developer-docs-site/docs/guides/multisig-multiagent-transaction.md

```diff
@@ -0,0 +1,296 @@
+    ty_args: [],
+    args: [
+      BCS.serialize(hexToAccountAddress("0x02")), // recipient of the transfer
+      BCS.serialize_uint64(2), // amount to transfer
+    ]
+  }
+
+  // Assume the multisigAccount has test coins
+  return {
+    "sender": multisigAccount,
+    "sequence_number": 1n,
+    "max_gas_amount": 2000n,
+    "gas_unit_price": 1n,
+    "expiration_timestamp_secs": Math.floor(Date.now() / 1000) + 600,
+    "payload": payload,
+    "chain_id": 3
+  };
+}
+```
+
+### Creating a signing message
+
+The below example code creates the signing message, following the same steps described in [Creating a Signed Transaction](guides/sign-a-transaction.md#step-2-creating-the-signing-message-and-signing-it).
+
+```typescript
+import * as Nacl from "tweetnacl";
+
+function hashPrefix(): Buffer {
+  let hash = SHA3.sha3_256.create();
+  hash.update(`APTOS::RawTransaction`);
+  return Buffer.from(hash.arrayBuffer());
+}
+
+function bcsSerializeRawTransaction(txn: RawTransaction): Buffer {
+  ...
+}
+
+// This will serialize a raw transaction into bytes
+function serializeRawTransaction(txn: RawTransaction): Buffer {
+  // Generate a hash prefix
+  const prefix = hashPrefix();
+
+  // Serialize txn with BCS
+  const bcsSerializedTxn = bcsSerializeRawTransaction(txn);
+
+  return Buffer.concat([prefix, bcsSerializedTxn]);
+}
+
+const rawTxn = createRawTransaction();
+const signingMessage = serializeRawTransaction(rawTxn);
+```
+
+### Creating a multisig Authenticator and a SignedTransaction
+
+A multisig transaction requires at least K (threshold) signers to sign the same signing message. See the below code example. 
+
+```typescript
+interface  MultiEd25519PublicKey {
+  // A list of public keys
+  public_keys: Uint8Array[],
+  // At least `threshold` signatures must be valid
+  threshold: Uint8,
+}
+
+interface MultiEd25519Signature {
+    // A list of signatures
+    signatures: Uint8Array[],
+    // 4 bytes, at most 32 signatures are supported.
+    // If Nth bit value is `1`, the Nth signature should be provided in `signatures`. Bits are read from left to right.
+    bitmap: Uint8Array,
+}
+
+interface MultisigAuthenticator {
+  public_key: MultiEd25519PublicKey,
+  signature: MultiEd25519Signature
+}
+
+const sig0 = aliceAccount.signBuffer(signingMessage);
+const sig2 = charlieAccount.signBuffer(signingMessage);
+// Marks the 1st and 3rd signer has signed. Left-most bit in first element is sig0 in this example.
+const bitmap = new Uint8Array([0b10100000, 0b00000000, 0b00000000, 0b00000000]);
+
+const muliEd25519Sig: MultiEd25519Signature = {
+  signatures: [
+    sig0.toUint8Array(),
+    sig2.toUint8Array(),
+  ],
+  bitmap
+};
+
+const multisigAuthenticator: MultisigAuthenticator = {
+  public_key: {
+    public_keys: [alicePubKey, bobPubKey, charliePubKey],
+    threshold: 2
+  },
+  signature: muliEd25519Sig
+}
+
+const signedTransaction: SignedTransaction = {
+  raw_txn: rawTxn,
+  authenticator: multisigAuthenticator
```

Comment:
my recollection is that the authenticator is opaque and can support either single or multisig?